src/preferences/preferences.cpp

```diff
@@ -1339,145 +1339,150 @@ void Preferences::disableRecursiveDownload(bool disable) {
 
 #ifdef Q_OS_WIN
 namespace {
-enum REG_SEARCH_TYPE {USER, SYSTEM_32BIT, SYSTEM_64BIT};
-
-QStringList getRegSubkeys(const HKEY &handle) {
-  QStringList keys;
-  DWORD subkeys_count = 0;
-  DWORD max_subkey_len = 0;
-  long res = ::RegQueryInfoKey(handle, NULL, NULL, NULL, &subkeys_count, &max_subkey_len, NULL, NULL, NULL, NULL, NULL, NULL);
-  if (res == ERROR_SUCCESS) {
-    max_subkey_len++; //For null character
-    LPTSTR key_name = new TCHAR[max_subkey_len];
-
-    for (uint i=0; i<subkeys_count; i++) {
-      res = ::RegEnumKeyEx(handle, 0, key_name, &max_subkey_len, NULL, NULL, NULL, NULL);
-      if (res == ERROR_SUCCESS)
-        keys.push_back(QString::fromWCharArray(key_name));
-    }
-    delete[] key_name;
-  }
+enum REG_SEARCH_TYPE
+{
+    USER,
+    SYSTEM_32BIT,
+    SYSTEM_64BIT
+};
 
-  return keys;
-}
+QStringList getRegSubkeys(HKEY handle)
+{
+    QStringList keys;
+
+    DWORD cSubKeys = 0;
+    DWORD cMaxSubKeyLen = 0;
+    LONG res = ::RegQueryInfoKeyW(handle, NULL, NULL, NULL, &cSubKeys, &cMaxSubKeyLen, NULL, NULL, NULL, NULL, NULL, NULL);
+
+    if (res == ERROR_SUCCESS) {
+        cMaxSubKeyLen++; // For null character
+        LPWSTR lpName = new WCHAR[cMaxSubKeyLen];
+        DWORD cName;
+
+        for (DWORD i = 0; i < cSubKeys; ++i) {
+            cName = cMaxSubKeyLen;
```

Comment:
>Prevent max subkey length overwrite.
    Can you point me to that code section?

It is here!
Next call to RegEnumKeyEx() will overwrite ```cName``` to actual size of lpName, so if we use ```cMaxSubKeyLen``` there it will be overwritten.